homeassistant/components/ovo_energy/sensor.py

```diff
@@ -32,199 +125,54 @@ async def async_setup_entry(
 
     if coordinator.data:
         if coordinator.data.electricity:
-            entities.append(OVOEnergyLastElectricityReading(coordinator, client))
-            entities.append(
-                OVOEnergyLastElectricityCost(
-                    coordinator,
-                    client,
-                    coordinator.data.electricity[
-                        len(coordinator.data.electricity) - 1
-                    ].cost.currency_unit,
-                )
+            entities.extend(
+                [
+                    OVOEnergySensor(coordinator, description, client)
+                    for description in SENSOR_TYPES_ELECTRICITY
+                ]
             )
         if coordinator.data.gas:
-            entities.append(OVOEnergyLastGasReading(coordinator, client))
-            entities.append(
-                OVOEnergyLastGasCost(
-                    coordinator,
-                    client,
-                    coordinator.data.gas[
-                        len(coordinator.data.gas) - 1
-                    ].cost.currency_unit,
-                )
+            entities.extend(
+                [
+                    OVOEnergySensor(coordinator, description, client)
+                    for description in SENSOR_TYPES_GAS
+                ]
             )
 
     async_add_entities(entities, True)
 
 
 class OVOEnergySensor(OVOEnergyDeviceEntity, SensorEntity):
```

Comment:
I'm wondering if `OVOEnergyDeviceEntity` and `OVOEnergyEntity` are necessary anymore. They don't do much assign from setting a few attributes. That could be done in `OVOEnergySensor` as well.

Especially since you inherit from it only once now.